src/providers/twitch/IrcMessageHandler.cpp

```diff
@@ -778,32 +343,41 @@ void IrcMessageHandler::handlePrivMessage(Communi::IrcPrivateMessage *message,
     }
 
     auto *twitchChannel = dynamic_cast<TwitchChannel *>(chan.get());
+    if (!twitchChannel)
+    {
+        return;
+    }
 
-    if (twitchChannel != nullptr)
+    parsePrivMessageInto(message, *twitchChannel, twitchChannel);
+}
+
+void IrcMessageHandler::parsePrivMessageInto(
+    Communi::IrcPrivateMessage *message, MessageSink &sink,
+    TwitchChannel *channel)
+{
+    auto currentUser = getApp()->getAccounts()->twitch.getCurrent();
+    if (message->tag("user-id") == currentUser->getUserId())
     {
-        auto currentUser = getApp()->getAccounts()->twitch.getCurrent();
-        if (message->tag("user-id") == currentUser->getUserId())
+        auto badgesTag = message->tag("badges");
+        if (badgesTag.isValid())
         {
-            auto badgesTag = message->tag("badges");
-            if (badgesTag.isValid())
-            {
-                auto parsedBadges = parseBadges(badgesTag.toString());
-                twitchChannel->setMod(parsedBadges.contains("moderator"));
-                twitchChannel->setVIP(parsedBadges.contains("vip"));
-                twitchChannel->setStaff(parsedBadges.contains("staff"));
-            }
+            auto parsedBadges = parseBadges(badgesTag.toString());
+            channel->setMod(parsedBadges.contains("moderator"));
+            channel->setVIP(parsedBadges.contains("vip"));
+            channel->setStaff(parsedBadges.contains("staff"));
```

Comment:
Not sure if I'm a fan of this running for historic messages, but I don't see how it could be that bad right now.